src/settings.cpp

```diff
@@ -274,7 +273,7 @@ static void MakeIntList(char *buf, const char *last, const void *array, int nele
 			case SLE_VAR_U32: v = *(const uint32 *)p; p += 4; break;
 			default: NOT_REACHED();
 		}
-		buf += seprintf(buf, last, (i == 0) ? "%d" : ",%d", v);
+		buf += seprintf(buf, last, IsSignedVarMemType(type) ? ((i == 0) ? "%d" : ",%d") : (type & SLF_HEX) ? ((i == 0) ? "%#X" : ",%#X") : ((i == 0) ? "%u" : ",%u"), v);
```

Comment:
explicit `0x` instead of `%#X` matches what was done in #7385

Reply:
also not a fan of the nested ternary, can it be split up?

Reply:
> explicit `0x` instead of `%#X` matches what was done in #7385

How do i know that wasn't just because that author didn't know about #?

Reply:
You didn't but you do now? :-)

Reply:
... I still don't know if you know what you know now.